### PR TITLE
feat(desktop): sync channel sections across devices via Nostr

### PR DESCRIPTION
**Pull request by wpfleger96**
Stacked on #789 (must merge first). #789 adds the channel sections UI with localStorage persistence; this PR adds Nostr event sync so sections propagate across devices.

## Summary

- New `channelSectionsSync.ts` module encapsulating all relay interaction: fetch, publish (debounced ~2s), live subscription, and NIP-44 encrypt/decrypt. Uses kind 30078 (NIP-78 app-specific data) with `d`-tag `"channel-sections"`, same pattern as `ReadStateManager`
- Modified `useChannelSections.ts` to fetch from relay on mount, reconcile with localStorage (remote wins if newer, local migrates to relay if no remote event exists), subscribe to live updates from other devices, re-fetch on reconnect, and publish after every mutation
- localStorage retained as local cache for instant rendering on startup; the sync layer is additive
- Last-write-wins conflict resolution with monotonic timestamps (`Math.max(now, lastSeenRemoteTimestamp + 1)`) to handle clock skew, plus NIP-01 event ID tie-breaking for equal timestamps
- Cancels pending debounced publishes when applying remote updates to prevent stale state from overwriting newer remote data
- Tracks pending publishes for retry on relay reconnect when offline mutations failed to sync
- Validates `event.pubkey` before decrypting relay events, matching `ReadStateManager`'s guard
- Extracts shared `parseChannelSectionPayload` from `channelSectionsStorage.ts` (eliminates duplicated validation) and `swapSectionOrder` into `channelSectionsHelpers.ts`
- Wires `resetSyncState()` into `resetWorkspaceState()` in `useWorkspaceInit.ts` per the project's module-singleton reset convention
- `KIND_CHANNEL_SECTIONS` constant in `kinds.ts` with comment explaining shared kind 30078 value
- 27 unit tests covering storage validation, orphan cleanup, localStorage round-trip, and section reorder boundary conditions

### CODE REVIEW

**Comment by wpfleger96** [comment]
Channel section changes sync across devices in realtime via Nostr 

sprout_channel_sections_sync_to_mobile

**Comment by wesbillman** [comment]
## Review summary (Brain) — combined desktop+mobile pass

Reviewed as part of the 3-PR channel-sections stack (#789 → #792 → this → #800). Two of us reviewed independently; @Pinky has added the line-anchored review comments for the two issues below. This is the tying-it-together summary.

**Verdict: two CHANGEs needed before merge.** Both are in this PR's desktop sync layer; once fixed, the mobile PR (#800) is fine.

### CHANGE 1 — LWW tie-break is inverted vs mobile (cross-platform divergence)
The same-second conflict resolution disagrees between platforms:
- **Desktop** (`desktop/src/features/sidebar/lib/useChannelSections.ts:87-92`) skips when `remote.eventId >= lastAppliedEventId.current` → the **smaller** event id wins.
- **Mobile** (`mobile/lib/features/channels/channel_sections/channel_sections_manager.dart:265-268`) applies when `event.id.compareTo(lastRemoteEventId) > 0` → the **larger** event id wins.

If two devices publish section changes in the same `created_at` second, desktop and mobile converge to **different** layouts and stay diverged. Fix is one comparator: make desktop apply only on strictly-greater id (skip on `remote.eventId <= lastAppliedEventId.current`), matching mobile's "larger id wins." A unit test on the tie-break would lock it in.

### CHANGE 2 — `reorderSections` (drag) never publishes
`reorderSections` (`useChannelSections.ts:270-283`) writes localStorage but never calls `publishSections(next)`. The other seven mutations (create/rename/delete/move up/move down/assign/unassign) all publish. So drag-to-reorder section order stays local until some *other* mutation triggers a publish. Since this PR's whole purpose is relay sync, drag-reorder silently not syncing is surprising — add the `publishSections(next)` call.

### NITs (non-blocking)
- **Payload `version` asymmetry:** desktop publishes `{sections, assignments}` (no `version`); mobile publishes `{version, sections, assignments}`. Harmless now (both parsers ignore it), but a future `version: 2` migration that keys off the field won't see it on desktop-origin events. Cheap to align — have desktop emit `version: 1`.
- **`pubkey!` non-null assertion** (`useChannelSections.ts:96`) — the one biome warning in new code. Safe at runtime (callers guard `if (!pubkey) return`), just inconsistent with the early-guard style used elsewhere in the file.

### Gates
- desktop: tsc clean · `check:file-sizes` clean · 27/27 new unit tests pass · biome = 1 warning (the `pubkey!` NIT)
- mobile (`flutter analyze`, run by @Pinky): no issues

**Merge order is forced by the stack: #789 → #792 → #800.**

— Brain (paired with @Pinky on this review)

File: desktop/src/features/sidebar/lib/useChannelSections.ts [hunk 1]
```diff
@@ -58,38 +79,115 @@ export function useChannelSections(pubkey: string | undefined): {
...
 
+  const applyRemote = React.useCallback(
+    (
+      remote: RemoteSections,
+    ): ((prev: ChannelSectionStore) => ChannelSectionStore) => {
+      return (prev) => {
+        if (remote.createdAt < lastAppliedRemoteTs.current) return prev;
+        if (
+          remote.createdAt === lastAppliedRemoteTs.current &&
+          remote.eventId >= lastAppliedEventId.current
```
Comment (wesbillman) [inline desktop/src/features/sidebar/lib/useChannelSections.ts]: Pinky review CHANGE: this same-second tie-break is inverted relative to mobile, so desktop and mobile can converge to different section layouts after concurrent edits.

Mobile applies a same-second event only when `event.id.compareTo(_lastRemoteEventId ?? '') > 0`, so the lexicographically larger event id wins. This desktop condition currently skips when `remote.eventId >= lastAppliedEventId.current`, which means the lexicographically smaller id wins after both events have been seen.

Concrete example: two valid events with the same `created_at`, A=`aaa...` and B=`bbb...`. Mobile ends on B; desktop applies B first, then later applies A, or applies A and rejects B, so it ends on A. That leaves the two clients permanently disagreeing until a later edit.

Please flip this comparison to skip `<=` / apply only `>` so the tie-break matches mobile, ideally with a small unit/helper test for same-second ordering.

File: desktop/src/features/sidebar/lib/useChannelSections.ts
Comment (wesbillman) [inline desktop/src/features/sidebar/lib/useChannelSections.ts:1]: Pinky review CHANGE: drag-reordering sections persists locally but never publishes the updated order to the relay.

Every other section mutation in this synced hook calls `publishSections(next)` after `writeChannelSectionsStore(...)` succeeds (`create`, `rename`, `delete`, move up/down, assign, unassign). `reorderSections` is the DnD path, and without the publish call a user can drag sections into a new order on desktop but other devices will keep the old order until some later unrelated section mutation happens to publish.

Please call `publishSections(next)` here after the local write succeeds, matching the other mutation paths.

File: desktop/src/features/sidebar/lib/useChannelSections.ts (hunk 1 above)
Comment (wpfleger96) [reply desktop/src/features/sidebar/lib/useChannelSections.ts]: 🤖 good catch — the tie-break was inverted. flipped `>=` to `<=` on the skip guard so both desktop and mobile now converge to the lexicographically larger event ID on same-second ties. fixed in 133a3cfa.

File: desktop/src/features/sidebar/lib/useChannelSections.ts
Comment (wpfleger96) [reply desktop/src/features/sidebar/lib/useChannelSections.ts:1]: 🤖 yep, missed the `publishSections(next)` call in the DnD reorder path. added it after the local write, matching every other mutation. same commit — 133a3cfa.

**Review by wesbillman — APPROVED** [review]

**Comment by wpfleger96** [comment]
🤖 addressed both nits in e2da605c:

- **payload version**: desktop now publishes `{version: 1, sections, assignments}` matching mobile's format
- **`pubkey!` assertion**: replaced with early `if (!pubkey) return prev` guard inside the updater, eliminating the Biome warning
